<launch>
    <arg name="arm_command_gui" default="false"/>
    <!-- Note that changing these from the default requires matching changes in the Java code -->
    <arg name="left_arm_recv_url" default="udp://10.10.10.108:30002"/>
    <arg name="right_arm_recv_url" default="udp://10.10.10.108:30001"/>

    <arg name="model" default="$(find-pkg-share victor_description)/urdf/victor.urdf.xacro"/>

    <!-- <include file="$(find-pkg-share victor_moveit_config)/launch/planning_context.launch.xml">
        <arg name="load_robot_description" value=""/>
    </include> -->

    <group ns="victor">
        <node pkg="victor_hardware_interface" exec="joint_state_publisher.py" name="joint_state_publisher"/>
        <node pkg="robot_state_publisher" exec="robot_state_publisher" name="robot_state_publisher"/>
        <node pkg="victor_hardware_interface" exec="arm_command_gui.py" name="arm_command_gui" if="$(var arm_command_gui)"/>
        <group ns="left_arm">
            <node pkg="victor_hardware_interface" exec="victor_hardware_interface_arm_wrapper_node"
                  name="left_arm_lcm_bridge_node" output="screen">
                <param name="cartesian_pose_frame" value="victor_left_arm_world_frame_kuka"/>
                <param name="send_lcm_url" value="udp://10.10.10.12:30000"/>
                <param name="recv_lcm_url" value="$(var left_arm_recv_url)"/>
            </node>
        </group>

        <group ns="right_arm">
            <node pkg="victor_hardware_interface" exec="victor_hardware_interface_arm_wrapper_node"
                  name="right_arm_lcm_bridge_node" output="screen">
                <param name="cartesian_pose_frame" value="victor_right_arm_world_frame_kuka"/>
                <param name="send_lcm_url" value="udp://10.10.10.11:30000"/>
                <param name="recv_lcm_url" value="$(var right_arm_recv_url)"/>
            </node>
        </group>

        <node pkg="joy" exec="joy_node" name="xbox_joystick"/>
        <node pkg="victor_hardware_interface" exec="xbox_control.py" name="xbox_to_victor">
            <param name="enable_finger_open_close_control" value="true"/>
            <param name="enable_scissor_open_close_control" value="true"/>
        </node>
        <node pkg="victor_hardware_interface" exec="arm_wrench_republisher.py" name="arm_wrench_republisher"
              output="screen"/>
        <node pkg="victor_hardware_interface" exec="grasp_status_node.py" name="grasp_status"/>
    </group>

</launch>
